### PR TITLE
Include filename and line number (closes #10)

### LINKED ISSUE #10: Include line number of failed go test
Including the line number of a failed go test will allow for https://github.com/ctrf-io/github-actions-test-reporter-ctrf to annotate correctly on Github.

Currently supported items are listed here: https://github.com/ctrf-io/go-ctrf-json-reporter?tab=readme-ov-file#test-object-properties

### PR DESCRIPTION
**Pull request by dmavrommatis**
Addressing https://github.com/ctrf-io/go-ctrf-json-reporter/issues/10

The failed tests usually have the filename and the line number at the start of the printed message.

This PR uses a regex matcher to find that information and add it to the generated CTRF json.

```
cat ctrf-report.json 
{
  "results": {
    "tool": {
      "name": "gotest"
    },
    "summary": {
      "tests": 1,
      "passed": 0,
      "failed": 1,
      "pending": 0,
      "skipped": 0,
      "other": 0,
      "start": 1728408248844,
      "stop": 1728408248844
    },
    "tests": [
      {
        "name": "Test",
        "status": "failed",
        "duration": 0,
        "suite": "command-line-arguments",
        "message": "=== RUN   Test\n    whatever_test.go:156: \n        \tError Trace:\t/home/dmavrommatis/workspace/whatever_test.go:156\n        \tError:      \tWomp womp\n        \tTest:       \tTest\n--- FAIL: Test (0.00s)\n",
        "trace": "whatever_test.go:156",
        "filepath": "whatever_test.go:156"
      }
    ]
  }
}
```

### CODE REVIEW

**Comment by dmavrommatis** [comment]
I see the trace and path being set correctly but I still do not see the annotation being created correctly. which field is passed to the annotation? trace or filepath?

and does it do any glob find in the repository to find the file or not?

File: reporter/reporter.go [hunk 1]
```diff
@@ -20,6 +21,8 @@ type TestEvent struct {
 	Output  string
 }
 
+var filenameMatcher = regexp.MustCompile(`([a-zA-Z0-9_/]+\.go):(\d+)`)
```
Comment (asherf) [inline reporter/reporter.go:24]: I think using a regex named group capture here will make the code more readable and easier to maintain over time.

**Comment by asherf** [comment]
readme needs to be updated to indicate we populate this new field.
https://github.com/ctrf-io/go-ctrf-json-reporter/blob/main/README.md#test-object-properties

**Comment by dmavrommatis** [comment]
will address the comments when I find some time

**Comment by dmavrommatis** [comment]
I tested this and the annotations created by the CTRF reporter afterwards are still not working correctly (maybe it will work with `-fullpath` but haven't tried). 

Unfortunately I do not have much time to work on this so I will close the PR for now until someone else picks it up.

**Pull request closed without merging**